### PR TITLE
[azure-iot-sdk-c] LTS_01_2023_REF01

### PR DESCRIPTION
**Pull request by JoseSanchez7**
[azure-iot-sdk-c] LTS_01_2023_REF01

### CODE REVIEW

**Comment by JoseSanchez7** [comment]
Hi @LilyWangLL , For umock-c and azure-macro-utils-c, we're still pointing to the same commit as before in our repos. So would it be okay not to update the ports for those, as I have done? Thanks.

**Comment by LilyWangLL** [comment]
> Hi @LilyWangLL , For umock-c and azure-macro-utils-c, we're still pointing to the same commit as before in our repos. So would it be okay not to update the ports for those, as I have done? Thanks.

We just need update ports which have new commit or release, if `umock-c` and `azure-macro-utils-c` have no new change, you don't need change them.

File: ports/azure-iot-sdk-c/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azure-iot-sdk-c",
-  "version-date": "2022-09-15",
+  "version-date": "2023-01-18",
   "port-version": 1,
```
Comment (LilyWangLL) [inline ports/azure-iot-sdk-c/vcpkg.json:4]:
```suggestion
```
Please remove `port-version`.

**Review by LilyWangLL — DISMISSED** [review]

**Comment by LilyWangLL** [comment]
Is the feature `public-preview` still needed? I found that when these `azure*` ports updating to new version, this feature has no change, and its dependencies also need to be handled.

**Comment by JoseSanchez7** [comment]
> Is the feature `public-preview` still needed? I found that when these `azure*` ports updating to new version, this feature has no change, and its dependencies also need to be handled.

Could you please clarify a bit more? Are you suggesting that the two options available are to either remove public-preview if it's no longer needed, and update vcpkg accordingly, or, if we're going to keep public-preview, then handle its dependencies?

**Comment by JoseSanchez7** [comment]
> Is the feature `public-preview` still needed? I found that when these `azure*` ports updating to new version, this feature has no change, and its dependencies also need to be handled.

We will be keeping the public-preview feature. So what it is we need to do now in order to be able to merge this PR?

**Comment by BillyONeal** [comment]
> > Is the feature `public-preview` still needed? I found that when these `azure*` ports updating to new version, this feature has no change, and its dependencies also need to be handled.
> 
> We will be keeping the public-preview feature. So what it is we need to do now in order to be able to merge this PR?

It seems that either the public preview should have its version updated to whatever the version after the one you're updating here is set to, or it should be outright removed. If you're editing the port yourself to change the ref you may as well just change the thing directly without the feature.

**Comment by ewertons** [comment]
Hi @BillyONeal , the public-preview switch in our vcpkg points to a specific branch that has not been updated, and it does not need to be updated.
Thus our understanding that the ref for public-preview in this vcpkg does not need to change.
And it must not be removed as well.

Could you clarify what is your request here, why are you saying we need to update the public-preview ref?

**Comment by BillyONeal** [comment]
> Hi @BillyONeal , the public-preview switch in our vcpkg points to a specific branch that has not been updated, and it does not need to be updated.

So there is no public preview change which contains the changes that match the official release you pushed here?

> Thus our understanding that the ref for public-preview in this vcpkg does not need to change. And it must not be removed as well.

If a feature is meaningless it should be removed. We try to leave deprecated features in place when there is a new feature with the same or similar effects to avoid needlessly breaking people, but if something is meaningless it's meaningless and we should remove it.

> Could you clarify what is your request here, why are you saying we need to update the public-preview ref?

Just don't want to create the unexpected situation where picking the "public-preview" feature makes you go *backwards*.

**Comment by JoseSanchez7** [comment]
> > Hi @BillyONeal , the public-preview switch in our vcpkg points to a specific branch that has not been updated, and it does not need to be updated.
> 
> So there is no public preview change which contains the changes that match the official release you pushed here?
> 
> > Thus our understanding that the ref for public-preview in this vcpkg does not need to change. And it must not be removed as well.
> 
> If a feature is meaningless it should be removed. We try to leave deprecated features in place when there is a new feature with the same or similar effects to avoid needlessly breaking people, but if something is meaningless it's meaningless and we should remove it.
> 
> > Could you clarify what is your request here, why are you saying we need to update the public-preview ref?
> 
> Just don't want to create the unexpected situation where picking the "public-preview" feature makes you go _backwards_.

The public preview features we have in the vcpkg are aligned with the Azure IoT guidance, and the current plan is for the feature to remain.

**Comment by dg0yt** [comment]
> > > Could you clarify what is your request here, why are you saying we need to update the public-preview ref?
> > 
> > 
> > Just don't want to create the unexpected situation where picking the "public-preview" feature makes you go _backwards_.
> 
> The public preview features we have in the vcpkg are aligned with the Azure IoT guidance, and the current plan is for the feature to remain.

https://github.com/microsoft/vcpkg/blob/340ce929d05c177fe17e0f4cb309fa0d61a6b74f/ports/azure-iot-sdk-c/portfile.cmake#L3-L12
https://github.com/Azure/azure-iot-sdk-c/commit/cb2e8d390df56ffa31d08ca0a79ab58ff96160cc: 2 years ago.

https://github.com/microsoft/vcpkg/blob/340ce929d05c177fe17e0f4cb309fa0d61a6b74f/ports/azure-iot-sdk-c/portfile.cmake#L13-L25
https://github.com/Azure/azure-iot-sdk-c/commit/9fa25c9f996f06b57132c73b95a251a0598d786b: 2 weeks ago.

Hard to believe that there is a "preview" aspect in this feature if it is not updated.

**Comment by LilyWangLL** [comment]
BTW, now feature `public-preview` install failed on vcpkg.

**Comment by BillyONeal** [comment]
> The public preview features we have in the vcpkg are aligned with the Azure IoT guidance, and the current plan is for the feature to remain.

OK but... why? Are you saying that the preview is actually "ahead" of this release or not?

File: versions/a-/azure-iot-sdk-c.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
 {
   "versions": [
+    {
+      "git-tree": "c0e747b46da1a6642cce7718381c89ee08e29a5c",
```
Comment (vaavva) [inline versions/a-/azure-iot-sdk-c.json:4]: I think you should be able to remove one of these to only have the most recent (with port-version 0)

**Review by ewertons — DISMISSED** [review]

**Review by BillyONeal — APPROVED** [review]
Thanks for the update!

**Pull request merged**